src/NUnitFramework/tests/Attributes/TimeoutTests.cs

```diff
@@ -167,6 +181,49 @@ public void TimeoutWithMessagePumpShouldAbort()
             Assert.That(result.ResultState, Is.EqualTo(ResultState.Failure));
             Assert.That(result.Message, Is.EqualTo("Test exceeded Timeout value of 500ms"));
         }
+#endif
+
+#if !THREAD_ABORT
+        [Timeout(50)]
+        public void TestTimeoutAndReportsTimeoutFailure()
+        {
+            while (true) { }
+        }
+
+        [Test, Timeout(100)]
+        public void TestTimeoutDoesNotStopCompletion()
+        {
+            Thread.Sleep(20);
+            Assert.True(true);
+        }
+
+        [Timeout(100)]
+        public void TestTimeoutWhichThrowsTestException()
+        {
+            throw new ArgumentException($"{nameof(ArgumentException)} was thrown.");
```

Comment:
What should happen if an exception is thrown in the thread after we've stopped waiting for it due to timeout? I feel like we should add that exception message as an additional error to the test's result even though we've already reported the timeout failure. We should then re-report the test status to the engine so that the exception message is reflected in Test Explorer etc right away. Would this make sense to do?

Maybe something for a future PR.

Reply:
I think we should leave this for a future PR. I don't think that this will occur very often, and I think this could be difficult to implement, so I would rather have this PR shipped (and also version 3.12 😄).